client/src/components/ChallengeCard.tsx

```diff
@@ -1,49 +1,43 @@
 import Card from "react-bootstrap/Card";
 import Button from "./Button.tsx";
 import { StarFill } from "react-bootstrap-icons";
-
-enum Difficulty {
-  easy = "easy",
-  medium = "medium",
-  hard = "hard",
-}
-
+import { ChallengeDetailsShort } from "../types/ChallengeDetailsShort.ts";
 // TODO: extract challenge into its own type
+import { Link } from "react-router-dom";
 
-type ChallengeCardProps = {
-  title: string;
-  description: string;
-  href: string;
-  difficulty: Difficulty;
-};
-
-const DIFFICULTY_TO_STARCOUNT = {
-  easy: 1,
-  medium: 2,
-  hard: 3,
-};
 
 function ChallengeCard({
   title,
-  description,
-  href,
+  generalDescription,
+  id,
   difficulty,
-}: ChallengeCardProps) {
+}: ChallengeDetailsShort) {
+
+  function getDifficulty(): JSX.Element[] {
+    const difficultyList: JSX.Element[] = [];
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-enum-comparison
+    for (let i = 0; i <= difficulty; i++) {
+      difficultyList.push(<StarFill key={i} />);
+    }
+    return difficultyList;
```

Comment:
What does this function do?
Get the difficulties of a challenge and then return the list? But is ther emore than one difficulty for one challenge?

Reply:
ill add documentation. It adds the number of stars depending on difficulty enum value. One start for easy, 2 medium, etc.

Reply:
I think the older version is more clear. Why are we extracting that into a function?

Reply:
I extracted it into a useCallback now. The old version was always giving only one star.